cmd/task/task.go

```diff
@@ -0,0 +1,50 @@
+package task
+
+import (
+	"kool-dev/kool/cmd/shell"
+	"sync"
+
+	"github.com/gookit/color"
+)
+
+// Runner holds logic for running tasks
+type Runner interface {
+	Run(string, func() error) error
+}
+
+// DefaultRunner holds data for running tasks
+type DefaultRunner struct {
+	out shell.OutputWriter
+}
+
+// NewRunner creates a new task runner instance
+func NewRunner() Runner {
+	return &DefaultRunner{shell.NewOutputWriter()}
+}
+
+// Run runs task
+func (r *DefaultRunner) Run(message string, closure func() error) (err error) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	r.out.Printf("%s ... ", message)
```

Comment:
this is a nice feature to start with, but we need to advise it only works for no-output tasks. If the closure() has some kind of output, this would get messy.

I would like to try exploring something like:

- We have a pipe for capturing the task actual output.
- While executing a task, we have something like:

```
Running "task"...
-------------------------------
(original task output... keeps writing here)
-------------------------------
... (animated char, something like an ASCII spinner with `-\|/-` replacing sequence, or something for that matter.
```


So the last output line is ALWAYS our "progress/running" spinner, while above it we keep displaying any eventual output the task could have.

Not sure how to architecture that, we may discuss further, maybe for a future iteration?

Reply:
For an eventual spinner: https://github.com/briandowns/spinner :D